components/lib/calendar/Calendar.js

```diff
@@ -320,11 +342,47 @@ export const Calendar = React.memo(React.forwardRef((props, ref) => {
             newViewDate.setFullYear(newYear);
         }
 
+        if (currentView === 'month') {
+            incrementYear();
+        }
+        else if (currentView === 'year') {
+            incrementDecade();
+        }
+
+
         updateViewDate(event, newViewDate);
 
         event.preventDefault();
     }
 
+    const populateYearOptions = (start, end) => {
+        let _yearOptions = [];
+
+        for (let i = start; i <= end; i++) {
+            yearOptions.push(i);
+        }
+
+        setYearOptions(_yearOptions)
+    }
+
+    const decrementYear = () => {
+        setCurrentYear(prevState => prevState - 1)
```

Comment:
Just double checking but when you call `setCurrentYear` i don't think the `currentYear` variable is updated immediately is it?  it might be better to use a local variable and then call `setCurrentYear(localVar)` and use the `localVar` everywhere in this method.

Same comment for the methods below...

Reply:
I agree.